test/metabase/query_processor_test/expressions_test.clj

```diff
@@ -449,6 +450,234 @@
                  mt/rows
                  ffirst))))))
 
+;;; +----------------------------------------------------------------------------------------------------------------+
+;;; |                                               LITERAL EXPRESSIONS                                              |
+;;; +----------------------------------------------------------------------------------------------------------------+
+
+(def ^:private standard-literal-expression-defs
+  {"empty"  [:value ""    {:base_type :type/Text}]
+   "foo"    [:value "foo" {:base_type :type/Text}]
+   "zero"   [:value 0     {:base_type :type/Integer}]
+   "12345"  [:value 12345 {:base_type :type/Integer}]
+   "float"  [:value 1.234 {:base_type :type/Float}]
+   "True"   [:value true  {:base_type :type/Boolean}]
+   "False"  [:value false {:base_type :type/Boolean}]})
+
+(def ^:private standard-literal-expression-refs
+  [[:expression "empty"]
+   [:expression "foo"]
+   [:expression "zero"]
+   [:expression "12345"]
+   [:expression "float"]
+   [:expression "True"]
+   [:expression "False"]])
+
+(def ^:private standard-literal-expression-column-refs
+  [[:field "empty" {:base_type :type/Text}]
+   [:field "foo"   {:base_type :type/Text}]
+   [:field "zero"  {:base_type :type/Integer}]
+   [:field "12345" {:base_type :type/Integer}]
+   [:field "float" {:base_type :type/Float}]
+   [:field "True"  {:base_type :type/Boolean}]
+   [:field "False" {:base_type :type/Boolean}]])
+
+(def ^:private standard-literal-expression-row-formats
+  [str str int int 3.0 mt/boolish->bool mt/boolish->bool])
+
+(def ^:private standard-literal-expression-row-formats-with-id
+  (into [int] standard-literal-expression-row-formats))
+
+(def ^:private standard-literal-expression-values
+  (map second (vals standard-literal-expression-defs)))
+
+(defmethod driver/database-supports? [::driver/driver ::expression-literals]
+  [driver _feature database]
+  (driver/database-supports? driver :expressions database))
+
+;; top-level literal expressions not yet supported for these drivers
+(doseq [driver [:mongo :oracle :redshift :sqlite :sqlserver :vertica]]
+  (defmethod driver/database-supports? [driver ::expression-literals]
+    [_driver _feature _database]
+    false))
```

Comment:
Having to duplicate the list of drivers that are exceptions around this feature is even smellier :(